src/dsp_tools/commands/excel2xml/excel2xml_lib.py

```diff
@@ -1333,6 +1334,55 @@ def make_text_prop(
     return prop_
 
 
+def _escape_reserved_chars(text: str) -> str:
+    """
+    From richtext strings (encoding="xml"), escape the reserved characters <, > and &,
+    but only if they are not part of a standard standoff tag or escape sequence.
+    The standard standoff tags allowed by DSP-API are documented here:
+    https://docs.dasch.swiss/2023.12.01/DSP-API/03-endpoints/api-v2/text/standard-standoff/
+
+    Args:
+        text: the richtext string to be escaped
+
+    Returns:
+        the escaped richtext string
+    """
+    allowed_tags = [
+        "a( [^>]+)?",  # <a> is the only tag that can have attributes
+        "p",
+        "em",
+        "strong",
+        "u",
+        "sub",
+        "sup",
+        "strike",
+        "h1",
+        "ol",
+        "ul",
+        "li",
+        "tbody",
+        "table",
+        "tr",
+        "td",
+        "br",
+        "hr",
+        "pre",
+        "cite",
+        "blockquote",
+        "code",
+    ]
+    allowed_tags_regex = "|".join(allowed_tags)
+    lookahead = rf"(?!/?({allowed_tags_regex})>)"
+    lookbehind = rf"(?<!</?({allowed_tags_regex}))"
+    illegal_lt = rf"<{lookahead}"
+    illegal_gt = rf"{lookbehind}>"
+    illegal_amp = r"&(?![#a-zA-Z0-9]+;)"
+    text = regex.sub(illegal_lt, "&lt;", text)
+    text = regex.sub(illegal_gt, "&gt;", text)
+    text = regex.sub(illegal_amp, "&amp;", text)
```

Comment:
to me, this feels extremely hard to read and understand. Maybe it would be worth going for a solution that has a bit more repetition, but instead is more readable?

Reply:
I have no idea how I could improve it...